examples/chef/common/stubs.cpp

```diff
@@ -347,22 +360,116 @@ void emberAfWakeOnLanClusterInitCallback(EndpointId endpoint)
 }
 #endif
 
-void ApplicationInit()
+/**
+ * This initializer is for the application having refrigerator on EP1 and child
+ * temperatureControlledCabinet endpoints on EP2 and EP3.
+ */
+void RefrigeratorTemperatureControlledCabinetInit()
 {
-    ChipLogProgress(NotSpecified, "Chef Application Init !!!");
-
-#ifdef MATTER_DM_PLUGIN_REFRIGERATOR_ALARM_SERVER
-    // set Parent Endpoint and Composition Type for an Endpoint
     EndpointId kRefEndpointId           = 1;
     EndpointId kColdCabinetEndpointId   = 2;
     EndpointId kFreezeCabinetEndpointId = 3;
-    SetTreeCompositionForEndpoint(kRefEndpointId);
-    SetParentEndpointForEndpoint(kColdCabinetEndpointId, kRefEndpointId);
-    SetParentEndpointForEndpoint(kFreezeCabinetEndpointId, kRefEndpointId);
-    // set TagList
-    SetTagList(kColdCabinetEndpointId, Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(refrigeratorTagList));
-    SetTagList(kFreezeCabinetEndpointId, Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(freezerTagList));
-#endif // MATTER_DM_PLUGIN_REFRIGERATOR_ALARM_SERVER
+    if (DeviceTypes::EndpointHasDeviceType(kRefEndpointId, DeviceTypes::kRefrigeratorDeviceId))
+    {
```

Comment:
nit: you could reduce indent complexity by inverting the if with `if (!EndpointHas...) { return; }`
Less indented code is easier to read.

Is it normal to have a refrigerator on EP1 withou t having cold/freezer cabinets? If so, maybe we should still log that we did not find them.